presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -246,10 +246,7 @@ public PlanWithProperties visitEnforceSingleRow(EnforceSingleRowNode node, Conte
         @Override
         public PlanWithProperties visitAggregation(AggregationNode node, Context context)
         {
-            HashSet<Symbol> partitioningRequirement = new HashSet<>(node.getGroupingSets().get(0));
-            for (int i = 1; i < node.getGroupingSets().size(); i++) {
-                partitioningRequirement.retainAll(node.getGroupingSets().get(i));
-            }
+            Set<Symbol> partitioningRequirement = ImmutableSet.copyOf(node.getGroupingKeys());
```

Comment:
I think this is more consistent with default presto behavior for joins and normal aggregations (with single grouping set), where presto use all join key columns/group by columns for partitioning. For example in:

```
SELECT * FROM tableA a, tableB b WHERE a.x = b.x AND a.y=b.y;
SELECT count(*) FROM tableA a GROUP BY a.x, a.y;
```
Why should we choose to partition only by `x` column? Currently presto will use both `x` and `y`. Now when we add empty grouping set:
```
SELECT count(*) FROM tableA a GROUP BY GROUPING SETS ((a.x, a.y), ());
```
why should we limit partitioning columns? To be consistent, we should use as many columns as possible.

IMO we could consider whether it is sometimes more efficient to drop some columns from partitioning, either to speed up partitioning or to avoid additional exchanges, but such discussion/optimization should be done independently of `ROLLUP`/`CUBE` implementation, because it is much more general issue. For example if we have aggregation on columns `a, b, c, d` and join on columns `a, b`, depending on the order of those operations, presto will add or omit an additional exchange between aggregation and join. This is completely random decision, which should be cost based.